LibTCPServer.emx

```diff
@@ -1212,35 +1221,96 @@ public:
 		str.append(result);
 		reply(str);
 	}
-
-private:
-	TCPServer_Actor& capsuleInst;
 };
 
-class LocalServerConnectionFactory: public TCPServerConnectionFactory
-	/// A factory for LocalServerConnection.
-{
+// Intercept incoming TCP requests and give them to the TCPServer for processing
+class ServerJob : public RTJob {
+	TCPServer_Actor* server;
+	const unsigned int maxSupportedClient;
+	unsigned int port;
+	long maxWaitForReply;
+	bool logToStdOut;
+	std::list<LocalServerConnection*> connections;
+
 public:
-	LocalServerConnectionFactory(TCPServer_Actor& context)
-		: capsuleInst(context)
-	{
+	ServerJob(TCPServer_Actor* s, unsigned int p, bool log, long maxWait, const unsigned int maxClient)
+	:server(s), port(p), logToStdOut(log), maxWaitForReply(maxWait), maxSupportedClient(maxClient) {}
+
+	virtual const char* name( void ) const override {
+		return "TCP Server Job";
 	}
-	
-	TCPServerConnection* createConnection(const StreamSocket& socket)
-	{
-		return new LocalServerConnection(socket, capsuleInst);
+
+	virtual void mainLoop( void ) override {
+		enter( RTJob::running );
+
+		RTIOMonitor ioMonitor;
+		RTTcpSocket listenSock;
+		listenSock.registerWith(&ioMonitor);
+		listenSock.setMaxPendingConnections(maxSupportedClient);
+		if (!listenSock.create() || !listenSock.listen(port))
+		{
+			std::cerr << "Failed to create TCP accept socket" << std::endl;
+			listenSock.close();
+			return;
+		}
+
+		do {
+			static const RTTimespec awhile(5L /* s */, 0L/*maxWaitForReply*1000*1000*/ /* ns */);
+
+			if (ioMonitor.wait(&awhile)) {
+				//Incoming connection
+
+				LocalServerConnection* lsc = new LocalServerConnection(*server, listenSock, logToStdOut);
+
+				if (!lsc->init()) {
+					delete lsc;
+					continue;
+				}
+
+				connections.push_back(lsc);
+			}
+			else {
+				// Ongoing connections should all have been replied to by now
+				while (!connections.empty()) {
+					LocalServerConnection* lsc = connections.front();
+					if (lsc->hasReplied) {
+						std::cout << "Reply detected. Removing LocalServerConnection" << std::endl;
```

Comment:
This printout was for internal use. At least we should check if user wants messages printed on stdout.

Reply:
Fixed by using logToStdOut config parameter.